File: mouse_compare/src/models/Review.ts
export interface StoredReview {
  id: string;
  userId: string;
  userName: string;
  mouseId: string;
  mouseName: string;
  text: string;
  rating: number;
  createdAt: Date;
}

const reviews: StoredReview[] = [];
let nextReviewId = 1;

export function createReview(userId: string, userName: string, mouseId: string, mouseName: string, text: string, rating: number): StoredReview {
  const review: StoredReview = {
    id: `review-${nextReviewId++}`,
    userId, userName, mouseId, mouseName,
    text, rating,
    createdAt: new Date(),
  };
  reviews.push(review);
  return review;
}

export function getAllReviews(): StoredReview[] {
  return [...reviews];
}

export function deleteReview(id: string): boolean {
  const idx = reviews.findIndex(r => r.id === id);
  if (idx === -1) return false;
  reviews.splice(idx, 1);
  return true;
}

File: project_compare/mouse_compare/backend/AspNetCoreAuth/AspNetCoreAuth/Models/Review.cs
using System.ComponentModel.DataAnnotations;

namespace AspNetCoreAuth.Models;

public class Review
{
    public int Id { get; set; }

    public string UserId { get; set; } = string.Empty;

    public string UserName { get; set; } = string.Empty;

    public string MouseId { get; set; } = string.Empty;

    public string MouseName { get; set; } = string.Empty;

    [Required]
    public string Text { get; set; } = string.Empty;

    [Range(1, 10)]
    public int Rating { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}
